File: src/main/java/net/botlify/anticaptchacom/response/solution/HCaptchaSolution.java
package net.botlify.anticaptchacom.response.solution;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;
import org.jetbrains.annotations.NotNull;

@ToString
@EqualsAndHashCode
public class HCaptchaSolution {

  /**
   * Token string required for interacting with the submit form on the target website.
   */
  @NotNull
  @Getter
  private String gRecaptchaResponse;

  /**
   * Output of "window.hcaptcha.getRespKey()" function when it is available.
   * Some websites use it for additional verification.
   */
  @NotNull
  @Getter
  private String respKey;

  /**
   * User-Agent of worker's browser. Use it when you submit the response token.
   */
  @NotNull
  @Getter
  private String userAgent;

}

File: src/main/kotlin/net/botlify/anticaptchacom/data/response/solution/HCaptchaSolution.kt
package net.botlify.anticaptchacom.data.response.solution

@Deprecated("Not implemented yet")
data class HCaptchaSolution(
    /**
     * Token string required for interacting with the submit form on the target website.
     */
    val gRecaptchaResponse: String,

    /**
     * Output of "window.hcaptcha.getRespKey()" function when it is available.
     * Some websites use it for additional verification.
     */
    val respKey: String,

    /**
     * User-Agent of worker's browser. Use it when you submit the response token.
     */
    val userAgent: String
)
